packages/shorebird_cli/lib/src/validators/shorebird_yaml_asset_validator.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:io';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/validators/validators.dart';
+import 'package:yaml/yaml.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+class ShorebirdYAMLAssetValidator extends Validator {
+  final String pubspecYamlPath = p.join('pubspec.yaml');
+
+  @override
+  String get description => 'pubspec.yaml contains shorebird.yaml in assets';
+
+  @override
+  bool canRunInCurrentContext() => _pubspecYamlFile?.existsSync() ?? false;
+
+  @override
+  String get incorrectContextMessage => '''
+The ${_pubspecYamlFile?.path ?? 'pubspec.yaml'} file does not exist.
+The command you are running must be run within a Flutter app project.''';
+
+  @override
+  Future<List<ValidationIssue>> validate() async {
+    final pubspecYamlFile = _pubspecYamlFile;
+    if (pubspecYamlFile == null || !pubspecYamlFile.existsSync()) {
+      return [
```

Comment:
Isn't this code unreachable since `canRunInCurrentContext` would return false if there's no `pubspec.yaml`?

Reply:
Simplified the logic a bit in the latest commit